grassroots-frontend/src/components/textField.tsx

```diff
@@ -0,0 +1,44 @@
+import { TextInput, TextInputProps } from "@mantine/core";
+import { JSX } from "react";
+import {
+  FieldValues,
+  useController,
+  UseControllerProps,
+} from "react-hook-form";
+
+// Inspired by https://github.com/aranlucas/react-hook-form-mantine/blob/master/src/TextInput/TextInput.tsx
+
+// Default value is optional in the extended types, but if you leave it out, react-form-hooks gets confused
+// when used in this way. Explicitly make it required.
+export type TextFieldProps<T extends FieldValues> = Omit<
+  UseControllerProps<T>,
+  "defaultValue"
+> &
+  Required<Pick<UseControllerProps<T>, "defaultValue">> &
+  Omit<TextInputProps, "value" | "defaultValue">;
+
+export function TextField<T extends FieldValues>(
+  props: TextFieldProps<T>,
```

Comment:
This type seems like it could be 

```
{
name: ...
defaultValue: ...
label: ...
}
```

Should `props` be passed into the `TextInput`?

Reply:
Yeah, the approach I took here is both broken and incorrect!

We probably want to narrow the API surface of this as much as possible, so taking TextFieldProps is incorrect.

Reply:
Fixed.